File: OneCodeTeam/How-to-use-SPPageStatusSetter-to-set-SharePoint-29422/[VB]-How to use SPPageStatusSetter to set SharePoint page status/VB/VBSharePointSetPageStatus/PageStatusEventReceiver/PageStatusEventReceiver.vb
'****************************** Module Header ******************************\
' Module Name:  PageStatusEventReceiver.vb
' Project:      VBSharePointSetPageStatus
' Copyright (c) Microsoft Corporation.
' 
' This sample will show you how to add page status to an application page
' and from list event receiver. 
' This is custom web part.
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL
' All other rights reserved.
'  
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************/
Imports System
Imports System.Security.Permissions
Imports Microsoft.SharePoint
Imports Microsoft.SharePoint.Utilities
Imports Microsoft.SharePoint.Workflow

Public Class PageStatusEventReceiver
    Inherits SPItemEventReceiver


    Private statusBarWebPart As StatusBarWebPart

    ''' <summary>
    ''' An item was added.
    ''' </summary>
    Public Overrides Sub ItemAdded(properties As SPItemEventProperties)
        MyBase.ItemAdded(properties)

        Dim web As SPWeb = properties.Web
        Dim oUrl As String = properties.List.DefaultViewUrl
        Dim coll As Microsoft.SharePoint.WebPartPages.SPLimitedWebPartManager = web.GetLimitedWebPartManager(oUrl, System.Web.UI.WebControls.WebParts.PersonalizationScope.[Shared])

        If coll.WebParts.Count > 1 Then
            statusBarWebPart = DirectCast(coll.WebParts(1), StatusBarWebPart)
            If statusBarWebPart IsNot Nothing Then
                statusBarWebPart.Message = "Item Added-vb"
                coll.SaveChanges(statusBarWebPart)
            End If
        Else
            statusBarWebPart = New StatusBarWebPart()
            statusBarWebPart.Message = "Item Added-vb"
            statusBarWebPart.Hidden = True
            coll.AddWebPart(statusBarWebPart, "Left", 1)
            coll.SaveChanges(statusBarWebPart)
        End If
    End Sub


    ''' <summary>
    ''' An item was updated.
    ''' </summary>
    Public Overrides Sub ItemUpdated(properties As SPItemEventProperties)
        MyBase.ItemUpdated(properties)

        Dim web As SPWeb = properties.Web
        Dim oUrl As String = properties.List.DefaultViewUrl
        Dim coll As Microsoft.SharePoint.WebPartPages.SPLimitedWebPartManager = web.GetLimitedWebPartManager(oUrl, System.Web.UI.WebControls.WebParts.PersonalizationScope.[Shared])

        If coll.WebParts.Count > 1 Then
            statusBarWebPart = DirectCast(coll.WebParts(1), StatusBarWebPart)
            If statusBarWebPart IsNot Nothing Then
                statusBarWebPart.Message = "Item Updated-vb"
                coll.SaveChanges(statusBarWebPart)
            End If
        Else
            statusBarWebPart = New StatusBarWebPart()
            statusBarWebPart.Message = "Item Updated-vb"
            statusBarWebPart.Hidden = True
            coll.AddWebPart(statusBarWebPart, "Left", 1)
            coll.SaveChanges(statusBarWebPart)
        End If
    End Sub

End Class

File: OneCodeTeam/How-to-use-SPPageStatusSetter-to-set-SharePoint-29422/[C#]-How to use SPPageStatusSetter to set SharePoint page status/C#/CSSharePointSetPageStatus/PageStatusEventReceiver/PageStatusEventReceiver.cs
/****************************** Module Header ******************************\
 * Module Name:  PageStatusEventReceiver.cs
 * Project:      CSSharePointSetPageStatus
 * Copyright (c) Microsoft Corporation.
 * 
 * This sample will show you how to add page status to an application page
 * and from list event receiver. 
 *
 * This source is subject to the Microsoft Public License.
 * See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL
 * All other rights reserved.
 * 
 * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
 * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
 * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using System;
using System.Security.Permissions;
using Microsoft.SharePoint;
using Microsoft.SharePoint.Utilities;
using Microsoft.SharePoint.Workflow;

namespace CSSharePointSetPageStatus.PageStatusEventReceiver
{
    /// <summary>
    /// List Item Events.
    /// </summary>
    public class PageStatusEventReceiver : SPItemEventReceiver
    {
        StatusBarWebPart.StatusBarWebPart statusBarWebPart;

        /// <summary>
        /// An item was added.
        /// </summary>
        public override void ItemAdded(SPItemEventProperties properties)
        {
            base.ItemAdded(properties);

            SPWeb web = properties.Web;
            string oUrl = properties.List.DefaultViewUrl;
            Microsoft.SharePoint.WebPartPages.SPLimitedWebPartManager coll = web.GetLimitedWebPartManager(oUrl, System.Web.UI.WebControls.WebParts.PersonalizationScope.Shared);

            if (coll.WebParts.Count > 1)
            {
                statusBarWebPart = (StatusBarWebPart.StatusBarWebPart)coll.WebParts[1];
                if (statusBarWebPart != null)
                {
                    statusBarWebPart.Message = "Item Added";
                    coll.SaveChanges(statusBarWebPart);               
                }
            }
            else
            {
                statusBarWebPart = new StatusBarWebPart.StatusBarWebPart();
                statusBarWebPart.Message = "Item Added";
                statusBarWebPart.Hidden = true;
                coll.AddWebPart(statusBarWebPart, "Left", 1); 
                coll.SaveChanges(statusBarWebPart);
            }           
        }

       
        /// <summary>
        /// An item was updated.
        /// </summary>
        public override void ItemUpdated(SPItemEventProperties properties)
        {
            base.ItemUpdated(properties);

            SPWeb web = properties.Web;
            string oUrl = properties.List.DefaultViewUrl;
            Microsoft.SharePoint.WebPartPages.SPLimitedWebPartManager coll = web.GetLimitedWebPartManager(oUrl, System.Web.UI.WebControls.WebParts.PersonalizationScope.Shared);

            if (coll.WebParts.Count > 1)
            {
                statusBarWebPart = (StatusBarWebPart.StatusBarWebPart)coll.WebParts[1];
                if (statusBarWebPart != null)
                {
                    statusBarWebPart.Message = "Item Updated";
                    coll.SaveChanges(statusBarWebPart);               
                }
            }
            else
            {
                statusBarWebPart = new StatusBarWebPart.StatusBarWebPart();
                statusBarWebPart.Message = "Item Updated";
                statusBarWebPart.Hidden = true;
                coll.AddWebPart(statusBarWebPart, "Left", 1);
                coll.SaveChanges(statusBarWebPart);            
            }           
        }
    }
}